src/tree/scaling/ScaleRuleGroupItem.ts

```diff
@@ -3,43 +3,83 @@
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/
 
-import { Revision, ScaleRule } from "@azure/arm-appcontainers";
-import { AzureSubscription } from "@microsoft/vscode-azureresources-api";
-import { ThemeIcon, TreeItemCollapsibleState } from "vscode";
+import { KnownActiveRevisionsMode, Revision, ScaleRule } from "@azure/arm-appcontainers";
+import type { TreeElementBase } from "@microsoft/vscode-azext-utils";
+import type { AzureSubscription, ViewPropertiesModel } from "@microsoft/vscode-azureresources-api";
+import * as deepEqual from "deep-eql";
+import { ThemeIcon, TreeItem, TreeItemCollapsibleState } from "vscode";
+import { ext } from "../../extensionVariables";
 import { localize } from "../../utils/localize";
-import { ContainerAppModel } from "../ContainerAppItem";
-import { RevisionsItemModel } from "../revisionManagement/RevisionItem";
-import { createScaleRuleItem } from "./ScaleRuleItem";
+import type { ContainerAppModel } from "../ContainerAppItem";
+import { RevisionDraftItem, RevisionsDraftModel } from "../revisionManagement/RevisionDraftItem";
+import type { RevisionsItemModel } from "../revisionManagement/RevisionItem";
+import { ScaleRuleItem } from "./ScaleRuleItem";
 
-export interface ScaleRuleGroupItem extends RevisionsItemModel {
-    scaleRules: ScaleRule[];
-}
+const scaleRulesLabel: string = localize('scaleRules', 'Scale Rules');
+
+export class ScaleRuleGroupItem implements RevisionsItemModel, RevisionsDraftModel {
+    static readonly contextValue: string = 'scaleRuleGroupItem';
+    static readonly contextValueRegExp: RegExp = new RegExp(ScaleRuleGroupItem.contextValue);
+
+    // Used as the basis for the view; can reflect either the original or the draft changes
+    private readonly scaleRules: ScaleRule[];
+
+    constructor(
+        readonly subscription: AzureSubscription,
+        readonly containerApp: ContainerAppModel,
+        readonly revision: Revision
+    ) {
+        if (this.hasUnsavedChanges()) {
```

Comment:
It may be easier to discuss this offline, but I'll write a short blurb and we can discuss further.

But I think `RevisionsDraftModel` should have a function to set the properties based on if it has unsaved changes or not. That way, the `ScaleRuleGroupItem` (and any other item that needs this logic) doesn't have to have the branching logic in its constructor.

`RevisionDraftModel` should also have two abstract methods: `setProperties` and `setPropertiesDraft` (or something like that-- maybe `setTreeItem`?) that does the assignments.

So I think that would look something like this:

```
// For tree items that depend on the container app's revision draft template
export abstract class RevisionsDraftModel {
    hasUnsavedChanges: () => boolean | Promise<boolean>;
    abstract function setProperties(): void;
    abstract function setPropertiesDraft(): void;
    constructor(...) {
        this.hasUnsavedChanges() ? this.setPropertiesDraft() : this.setProperties();
    }
}

Reply:
Ran into some issues here with implementing it this way.  `hasUnsavedChanges` usually returns a promise that needs to be awaited.  But you can't async/await in a constructor... potentially making this way much less straightforward to implement

Reply:
After some testing, ran into some limitations with doing it this way, will discuss more offline